code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt

```diff
@@ -722,12 +722,34 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
     private fun publishAnalyticsId(event: Event) {
         val data = getSharedState()
         api.createSharedState(data, event)
-        val responseEvent = Event.Builder(
+
+        // dispatch paired response event, usually used for getters
+        val pairedResponseEvent = Event.Builder(
```

Comment:
Does it make sense to have 
`publishAnalyticsId(event: Event, paired: Bool = false) ` and call with paired = true for getters? That way we publish 1 instead of 2 events.

Reply:
seems like in all the situations we used to send both the events